File: ruby/reverse_string.rb
def reverse_string(str)
  new_array = []
  str_array = str.split('')
  $i = str_array.length
  while $i >= 0 do
    new_array << str_array[$i]
    $i -= 1
  end
  new_array.join('')
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')

  # it "can reverse multiple words" do
  #   expect(reverse_string('bla bla bla')).to eq('alb alb alb')
  # end

end

# Please add your pseudocode to this file
# And a written explanation of your solution

File: javascript/reverse_string.js
function reverseString(str) {
  const newArray = []
  const strArray = str.split('')
  let i = strArray.length
  while (i >= 0) {
    newArray.push(strArray[i])
    i -= 1
  }
  return newArray.join('')
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));
}

module.exports = reverseString;

// Please add your pseudocode to this file
// And a written explanation of your solution
